fastly/automation_token.go

```diff
@@ -0,0 +1,185 @@
+func (c *Client) GetAutomationTokens(i *GetAutomationTokensInput) *ListPaginator[AutomationTokenPaginator] {
+	input := ListOpts{}
+	if i.Page != nil {
+		input.Page = *i.Page
+	}
+	if i.PerPage != nil {
+		input.PerPage = *i.PerPage
+	}
+	return NewPaginator[AutomationTokenPaginator](c, input, "/automation-tokens")
+}
+
+// ListAutomationTokens retrieves all resources. Not suitable for large collections.
+func (c *Client) ListAutomationTokens() ([]*AutomationToken, error) {
+	p := c.GetAutomationTokens(&GetAutomationTokensInput{})
+	var results []*AutomationToken
+	for p.HasNext() {
+		data, err := p.GetNext()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get next page (remaining: %d): %s", p.Remaining(), err)
+		}
+
+		for _, t := range data {
+			results = append(results, t.Data...)
+		}
+	}
+	return results, nil
+}
+
+// GetAutomationTokenInput is used as input to the GetAutomationToken function.
+type GetAutomationTokenInput struct {
+	// TokenID is an alphanumeric string identifying the token (required).
+	TokenID string
+}
+
+// GetAutomationToken retrieves a specific resource by ID.
+func (c *Client) GetAutomationToken(i *GetAutomationTokenInput) (*AutomationToken, error) {
+	if i.TokenID == "" {
+		return nil, ErrMissingTokenID
+	}
+
+	path := ToSafeURL("automation-tokens", i.TokenID)
+
+	resp, err := c.Get(path, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var t *AutomationToken
+	if err := decodeBodyMap(resp.Body, &t); err != nil {
+		return nil, err
+	}
+	return t, nil
+}
+
+// CreateAutomationTokenInput is used as input to the CreateAutomationToken function.
+type CreateAutomationTokenInput struct {
+	// ExpiresAt is a time-stamp (UTC) of when the token will expire.
+	ExpiresAt time.Time `json:"expires_at" url:"expires_at,omitempty"`
+	// Name is the name of the token.
+	Name string `json:"name" url:"name,omitempty"`
+	// Password is the token password.
+	Password *string `json:"-" url:"password,omitempty"`
+	// Role is the role on the token (billing, engineer, user).
+	Role AutomationTokenRole `json:"role" url:"role,omitempty"`
+	// Scope is a space-delimited list of authorization scope (global, purge_select, purge_all, global).
+	Scope *TokenScope `json:"scope,omitempty" url:"scope,omitempty"`
+	// Services is a list of alphanumeric strings identifying services.
+	// If no services are specified, the token will have access to all services on the account.
+	Services []string `json:"services" url:"services,omitempty,brackets"`
+	// Username is the email of the user the token is assigned to.
+	Username *string `json:"-" url:"username,omitempty"`
+	// TLSAccess indicates whether TLS access is enabled for the token.
+	TLSAccess bool `json:"tls_access" url:"tls_access,omitempty"`
+}
+
+// CreateAutomationToken creates a new resource.
+func (c *Client) CreateAutomationToken(i *CreateAutomationTokenInput) (*AutomationToken, error) {
+	_, err := c.PostForm("/sudo", i, nil)
```

Comment:
This was also copy-and-paste from the regular token code, but the doc for this function should indicate that `sudo` capability must be available for the token that is used for this function.

Reply:
This has been updated to include a note about `sudo` being required.